gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/persistence/BackupManager.java

```diff
@@ -165,8 +164,29 @@ private File checkBaseline(File baselineParentDir) throws IOException{
     
     return baselineDir;
   }
-  
-  public HashSet<PersistentID> finishBackup(File targetDir, File baselineDir) throws IOException {
+
+  private Collection<DiskStoreImpl> getDiskStoresToBackUp(byte diskStoresToBackup) {
+    Collection<DiskStoreImpl> allDses = cache.listDiskStoresIncludingRegionOwned();
+    switch (diskStoresToBackup) {
+      case FinishBackupRequest.DISKSTORE_ALL:
+        return allDses;
+
+      case FinishBackupRequest.DISKSTORE_ALL_BUT_DD:
+        allDses.removeIf(ds -> ds.getName().equals("GFXD-DD-DISKSTORE"));
+        return allDses;
+
+      case FinishBackupRequest.DISKSTORE_DD:
+        Optional<DiskStoreImpl> ods = allDses.stream().filter
+            (ds -> ds.getName().equals("GFXD-DD-DISKSTORE")).findFirst();
+        if (ods.isPresent()) {
+          return Collections.singletonList(ods.get());
+        }
+    }
+    return Collections.emptyList();
+  }
+
+  public HashSet<PersistentID> finishBackup(File targetDir,
```

Comment:
We can make two APIs, one which takes disk store name, other one being the same as previous. The unchanged on will default to DISK_STORE_ALL. We don't have to change existing APIs.

Reply:
That is the reason why I sent an email for opinions that should we have a different disk store to be configured by user for first backup or internal one is fine. Since I did not get response for that, so I implemented in this way, which according to me is fine. Giving control to the user to specify which DiskStore should be backed up first and last probably is not a great idea, unless we see more use cases. If there are more use cases like that we can extend later. Right now I think using DD disktore is fine for the offset table purpose so it is fine.

Reply:
Maybe we can have internal API but I support not exposing to user. User's app shouldn't depend on which disk store should be backed up first so better not give any option.

Reply:
Yes, I was talking about internal APIs only. External APIs should not be concerned about order an all.

Reply:
@rishitesh and @suranjan, I see what you are saying. Mainly for testing it? and extend it to product tool option if required?
I think we can do it and the two phase written will help it to be done very quickly. However if you guys don't see any immediate use case of this then I would like to stick with what it is right now and quickly revisit if required.